shared/src/main/scala/bloop/io/AbsolutePath.scala

```diff
@@ -21,6 +21,7 @@ final class AbsolutePath private (val underlying: Path) extends AnyVal {
   def exists: Boolean = Files.exists(underlying)
   def isFile: Boolean = Files.isRegularFile(underlying)
   def isDirectory: Boolean = Files.isDirectory(underlying)
+  def isSameFile(other: AbsolutePath): Boolean = Files.isSameFile(underlying, other.underlying)
```

Comment:
What happens if this throws? Before we were only comparing strings so we didn’t need to worry about this. It would be handy if we report this error explicitly to the user with a runtime exception

Reply:
I added `isSameFile` in `CompilerCache`. WDYT?